src/server/game/Entities/GameObject/GameObject.cpp

```diff
@@ -2321,6 +2348,27 @@ void GameObject::SetLootState(LootState state, Unit* unit)
     }
 }
 
+void GameObject::SendLootStateUpdateToNearbyPlayers()
```

Comment:
What exactly is the point of this function here

Reply:
Gameobjects don't send update field changes when their loot state changes, this causes some issues like gameobjects being still active after a player loots the last quest item. This forces an update to all nearby players so they see the gameobject as activable or not activable depending on state.

Reply:
Just call `AddToObjectUpdateIfNeeded()` to schedule the update at the end of Map::Update, you are talking about GAMEOBJECT_DYNAMIC field which is always sent (if object gets its update sent), don't force immediate packet

Reply:
Fixed, thanks!